diff --git a/ihp-sg13g2/libs.tech/magic/ihp-sg13g2-GDS.tech b/ihp-sg13g2/libs.tech/magic/ihp-sg13g2-GDS.tech
index 454188a1..1febc741 100644
--- a/ihp-sg13g2/libs.tech/magic/ihp-sg13g2-GDS.tech
+++ b/ihp-sg13g2/libs.tech/magic/ihp-sg13g2-GDS.tech
@@ -1019,7 +1019,7 @@ style patternfill variants (),(tiled)
  templayer	met2fill_coarse topbox
         # slots   0 5000 420 0 5000 420 1700 850
 	# Make spacing 2.00um so that max density is < 50%
-        slots   0 5000 2000 0 5000 2000 2500 1500
+        slots   0 5000 840 0 5000 840 2500 1500
         and-not obstruct_m2
 	and	topbox
 	orthogonal
@@ -1047,16 +1047,91 @@ style patternfill variants (),(tiled)
         grow    420
 
  templayer	met2fill_fine topbox
-        slots   0 1200 420 0 1200 420 300 0
+        slots   0 1200 300 0 1200 300 300 0
         and-not obstruct_m2_fine
 	and	topbox
 	orthogonal
         shrink  595
         grow    595
 
+ templayer      obstruct_m2_finer BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM2,MET2BLOCK,MET2FILL,FILLBLOCK
+	or	met2fill_coarse,met2fill_medium,met2fill_fine
+        grow    420
+
+ templayer	met2fill_finer topbox
+        slots   0 1000 220 0 1000 220 0 0
+        and-not obstruct_m2_finer
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m2_finer2 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM2,MET2BLOCK,MET2FILL,FILLBLOCK
+	or	met2fill_coarse,met2fill_medium,met2fill_fine,met2fill_finer
+        grow    420
+
+ templayer	met2fill_finer2 topbox
+        slots   0 1000 420 0 1000 420 300 200
+        and-not obstruct_m2_finer2
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m2_finer3 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM2,MET2BLOCK,MET2FILL,FILLBLOCK
+	or	met2fill_coarse,met2fill_medium,met2fill_fine,met2fill_finer,met2fill_finer2
+        grow    420
+
+ templayer	met2fill_finer3 topbox
+        slots   0 1000 420 0 1000 420 300 400
+        and-not obstruct_m2_finer3
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m2_finer4 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM2,MET2BLOCK,MET2FILL,FILLBLOCK
+	or	met2fill_coarse,met2fill_medium,met2fill_fine,met2fill_finer,met2fill_finer2,met2fill_finer3
+        grow    420
+
+ templayer	met2fill_finer4 topbox
+        slots   0 1000 420 0 1000 420 300 600
+        and-not obstruct_m2_finer4
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m2_finer5 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM2,MET2BLOCK,MET2FILL,FILLBLOCK
+	or	met2fill_coarse,met2fill_medium,met2fill_fine,met2fill_finer,met2fill_finer2,met2fill_finer3,met2fill_finer4
+        grow    420
+
+ templayer	met2fill_finer5 topbox
+        slots   0 1000 420 0 1000 420 300 800
+        and-not obstruct_m2_finer5
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
  layer	MET2FILL met2fill_coarse
 	or met2fill_medium
 	or met2fill_fine
+	or met2fill_finer
+	or met2fill_finer2
+	or met2fill_finer3
+	or met2fill_finer4
+	or met2fill_finer5
  	calma 	10 22
 
 #---------------------------------------------------
@@ -1107,9 +1182,85 @@ style patternfill variants (),(tiled)
         shrink  595
         grow    595
 
+ templayer      obstruct_m3_finer BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM3,MET3BLOCK,MET3FILL,FILLBLOCK
+	or	met3fill_coarse,met3fill_medium,met3fill_fine
+        grow    420
+
+ templayer	met3fill_finer topbox
+        slots   0 1000 420 0 1000 420 0 0
+        and-not obstruct_m3_finer
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m3_finer2 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM3,MET3BLOCK,MET3FILL,FILLBLOCK
+	or	met3fill_coarse,met3fill_medium,met3fill_fine,met3fill_finer
+        grow    420
+
+ templayer	met3fill_finer2 topbox
+        slots   0 1000 420 0 1000 420 300 200
+        and-not obstruct_m3_finer2
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m3_finer3 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM3,MET3BLOCK,MET3FILL,FILLBLOCK
+	or	met3fill_coarse,met3fill_medium,met3fill_fine,met3fill_finer,met3fill_finer2
+        grow    420
+
+ templayer	met3fill_finer3 topbox
+        slots   0 1000 420 0 1000 420 300 400
+        and-not obstruct_m3_finer3
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m3_finer4 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM3,MET3BLOCK,MET3FILL,FILLBLOCK
+	or	met3fill_coarse,met3fill_medium,met3fill_fine,met3fill_finer,met3fill_finer3,met3fill_finer3
+        grow    420
+
+ templayer	met3fill_finer4 topbox
+        slots   0 1000 420 0 1000 420 300 600
+        and-not obstruct_m3_finer4
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
+ templayer      obstruct_m3_finer5 BIPOLARID,PADID,EDGESEAL
+	grow	580
+	or	ALLM3,MET3BLOCK,MET3FILL,FILLBLOCK
+	or	met3fill_coarse,met3fill_medium,met3fill_fine,met3fill_finer,met3fill_finer2,met3fill_finer3,met3fill_finer4
+        grow    420
+
+ templayer	met3fill_finer5 topbox
+        slots   0 1000 420 0 1000 420 300 800
+        and-not obstruct_m3_finer5
+	and	topbox
+	orthogonal
+        shrink  495
+        grow    495
+
  layer	MET3FILL met3fill_coarse
 	or	met3fill_medium
 	or	met3fill_fine
+	or met3fill_finer
+	or met3fill_finer2
+	or met3fill_finer3
+	or met3fill_finer4
+	or met3fill_finer5
+
  	calma 	30 22
 
 #---------------------------------------------------
